File: apps/python-service/app/routes/health.py
"""Health check routes."""

from datetime import datetime

from fastapi import APIRouter

router = APIRouter()


@router.get("/")
async def health_check():
    """Basic health check endpoint."""
    return {
        "success": True,
        "data": {
            "status": "healthy",
            "timestamp": datetime.utcnow().isoformat(),
            "service": "python-service",
        },
    }


@router.get("/ready")
async def readiness_check():
    """Readiness check endpoint."""
    # Add database connectivity check here
    return {
        "success": True,
        "data": {
            "status": "ready",
            "timestamp": datetime.utcnow().isoformat(),
        },
    }

File: apps/oauth-service/src/routes/health.ts
import { Router } from 'express';
import { successResponse } from '@mono/shared';
import { db } from '@mono/db';
import { sql } from 'drizzle-orm';

const router: Router = Router();

router.get('/', (_req, res) => {
  res.json(
    successResponse({
      status: 'healthy',
      timestamp: new Date().toISOString(),
      service: 'oauth-service',
    })
  );
});

router.get('/ready', async (_req, res) => {
  try {
    //doning simple command to check the database is respond
    await db.execute(sql`SELECT 1`);

    res.status(200).json(
      successResponse({
        status: 'ready',
        database: 'connected',
        timestamp: new Date().toISOString(),
      })
    );
  } catch (error) {
    console.error('Database connection error in health check:', error);
    
    //if there is error inside database
    res.status(503).json({
      success: false,
      data: {
        status: 'unready',
        database: 'disconnected',
        timestamp: new Date().toISOString(),
      }
    });
  }
});

export { router as healthRouter };
